### ISSUE TITLE
Title not centered in navBar in Android with one of left/right icons

### ISSUE THREAD

**Issue by iuraya** [RNRF/react-native-router-flux#2467]
### Version
Tell us which versions you are using: 

- react-native-router-flux v4.0.0-beta.21
- react-native v0.47.1

### Expected behaviour
Title appears correctly centered when neither left and right icons exists or when both of them exist
With just one of left or right icons at the same time title should appear centered

### Actual behaviour
The title is not centered if only one of right and left icons exists at the same time.

<img width="390" alt="captura de pantalla 2017-10-06 a las 16 47 20" src="https://user-images.githubusercontent.com/14769734/31289237-63b3e1c8-aac8-11e7-9ffb-1c262f748e0a.png">

**Comment by JessieYF** [comment]
I encountered this issue as well. What I do is adding a placeholder for left button and right button on root Scene. When the child Scene has any left/right button, override the property.

```
renderPlaceholder = () => <View />
<Scene
     key="root"
     renderLeftButton={this.renderPlaceholder}
     renderRightButton={this.renderPlaceholder}
/ >
```

eg: child Scene
```
<Scene
     key="sub"
     title="sub"
     component={PageComponent}
     hideTabBar
     renderLeftButton={() => <BackButton />}
/>
```

**Comment by aksonov** [comment]
Please check latest master and PR #2491 (I've just merged it). Unfortunately I'm working on ios app only.

**Comment by daviscabral** [comment]
@aksonov I can help you testing changes on android - just tag me in the PR and give me instructions and I can check it there. :+1:

**Comment by aksonov** [comment]
@daviscabral Could you check this ticket? :)

**Comment by sagark1510** [comment]
@aksonov it's still the same on android. Do I have to add any props to scene to get reflection or it should work automatically?

**Comment by aksonov** [comment]
Right, I see this bug and PR is welcome.

**Comment by PiotrKujawa** [comment]
<img width="403" alt="android_title_alignment" src="https://user-images.githubusercontent.com/4909699/34606549-2809f95c-f210-11e7-80d8-f13b9b469735.png">

**Comment by Albinzr** [comment]
Will this be fixed?,I cant push the app to production without a fix.

**Comment by kamranyounis12** [comment]
@daviscabral  any update?

**Comment by daviscabral** [comment]
Issue is still present on master - going to investigate now to see if I am able to solve it.
<img width="427" alt="screenshot 2018-02-23 06 34 54" src="https://user-images.githubusercontent.com/4285/36587346-ad0bc774-1863-11e8-826a-5b076cfde647.png">

**Comment by Gulci** [comment]
This issue is also present on React Native v. 0.54.2 with react-native-router-flux v. 4.0.0-beta.28.

**Comment by daviscabral** [comment]
I am a bit busy lately and couldn't finish the PR that I've started. I'll check if later today I'll spare some time for it. Thank you.

**Comment by daviscabral** [comment]
On it - should have a PR in a few hours. I am in a company transition, things are just crazy lately.

**Comment by daviscabral** [comment]
Sorry - took longer than I thought cause I was trying to fix the `Example` project that was not running - so I couldn't actually test and verify it properly.

**Comment by acollazomayer** [comment]
@daviscabral Thank you! I manage to solve the issue making a custom nav bar. However this is better! Thanks!! Good luck with the company transition!

**Comment by daviscabral** [comment]
Thank you. Glad to hear that.

**Comment by xbogdan** [comment]
I'm still having this issues with react-native 0.55 and react-native-router-flux 4.0.0-beta.31 .
Was this fixed or should I follow @acollazomayer solution ?

**Comment by massimopibiri** [comment]
The solution written by @JessieYF works good

**Comment by aksonov** [comment]
Please try to reproduce it with Example project and latest version 4.0.0-beta.40. Feel free to open if the issue still exists

**Comment by og24715** [comment]
This issue still exists with RNRF v4.0.5. Im not able to open it, but I think that it have to be open.

**Comment by aksonov** [comment]
cc @daviscabral

**Comment by daviscabral** [comment]
> Fixed on #3256.
> 
> [![Image from Gyazo](https://camo.githubusercontent.com/9d67376b43d915a2a6b3da600e5154cc9a7e318c/68747470733a2f2f692e6779617a6f2e636f6d2f38343939306530343335366564333837636562353063613939643434386135612e676966)](https://gyazo.com/84990e04356ed387ceb50ca99d448a5a)

This was possible because of react-navigation changes,  when `headerLayoutPreset` support was added. Docs were also updated to reflect it. 

`Example.js` file has an example of how to make it work, below the line where it happens:

```
<Stack key="customNavBar" hideTabBar headerLayoutPreset='center'>
```
